polkadot/runtime/parachains/src/configuration.rs

```diff
@@ -377,7 +377,7 @@ where
 			})
 		}
 
-		if self.max_pov_size > MAX_POV_SIZE {
+		if self.max_pov_size as u64 > sc_network::MAX_RESPONSE_SIZE {
```

Comment:
What if someone adds in the future a totally unrelated protocol and they decide to increase response_size to 48MiB or more because they need that, then you suddenly and accidentally allow all parachains PoVs to be 48MiBs large which might not be supported or safe.

Reply:
You suddenly and accidentally allow governance to set that value for the PoV size. I hope governance will not do it without analyzing the consequences :shrug:

Reply:
The check is only for setting the config value, so you'd still need governance to pass any change that will affect what all paras can do. 

However I also don't like bringing `sc-network` to this party. Why does the runtime need to depend on some networking configuration parameters? If no one has a better idea I suggest we promote this constant to a runtime configuration parameter.

Reply:
Correct missed the governance part :D

Reply:
@sandreim @alexggh how do you feel about the following changeset:
* Move `MAX_RESPONSE_SIZE` from `sc-network` to `sc-network-common`, a small crate with commonly used networking-related types;
* Move `POV_RESPONSE_SIZE` from `polkadot-node-network-protocol` to `polkadot-primitives`, which is already packed with a bunch of random low-level constants shared between node and runtime.

New dependencies:
* `polkadot-primitives` depends on `sc-network-common` (which sounds logical; the Polkadot protocol inherently depends on some network limitations)
* `polkadot-node-network-protocol` depends on `polkadot-primitives` (obviously not a problem);
* Everything other in `polkadot/` already depends on `polkadot-primitives`.

Reply:
I think it is fine for the node to depend on runtime stuff, but not the other way around. 

In the future we will want improve the `PoV` fetching mechanism both for collators and validators and we could go the way of streaming: https://github.com/paritytech/polkadot-sdk/issues/968.  This doesn't require to have the maximum notification size equal to the maximum PoV size since we'll chunk the PoV up to smaller pieces.

Reply:
> I'm just not quite sure if it is worth it because of a single constant

My2Cents: It is not worth it, we can just define a local constant here MAX_POV_RESONABLE_SIZE: 16MiB, you can write it some documentation why you can't go higher than 16MiB without increasing the network constant. 

I don't think we get too much benefit from using a single constant everywhere, because you still need to be aware of the different deployment cycles for node and runtime.

Reply:
I would go with a local constant in that case. We can add a comment why we don't use a constant from network here.